src/engine/SCons/EnvironmentTests.py

```diff
@@ -2035,7 +2038,7 @@ def __call__(self, command):
             assert save_command == ['fake command'], save_command
             assert env['ASFLAGS'] == ['assembler', '-as'], env['ASFLAGS']
             assert env['CCFLAGS'] == ['', '-X', '-Wa,-as',
-                                      '-pthread', '-mno-cygwin',
+                                      '-pthread', '-fmerge-all-constants', '-mno-cygwin',
```

Comment:
...and here you expect it to be seen before `-mno-cygwin`

Reply:
ugh! my bad! i didn't run the tests! (and i forgot that i could easily commit this suggestion :'( n00b...